salt/etcd-proxy/etcd-proxy.conf.jinja

```diff
@@ -4,19 +4,19 @@
 ETCD_DATA_DIR="/var/lib/etcd/"
 ETCD_NAME="{{ grains['id'] }}"
 
-ETCD_LISTEN_CLIENT_URLS="http://0.0.0.0:2379"
+ETCD_LISTEN_CLIENT_URLS="https://0.0.0.0:2379"
 ETCD_LISTEN_PEER_URLS="https://0.0.0.0:2380"
 ETCD_ADVERTISE_CLIENT_URLS="http://{{ grains['caasp_fqdn'] }}:2379"
 
-ETCD_CA_FILE={{ pillar['paths']['ca_dir'] }}/{{ pillar['paths']['ca_filename'] }}
-ETCD_CERT_FILE=/etc/pki/minion.crt
-ETCD_KEY_FILE=/etc/pki/minion.key
-ETCD_TRUSTED_CA_FILE={{ pillar['paths']['ca_dir'] }}/{{ pillar['paths']['ca_filename'] }}
+ETCD_CA_FILE={{ pillar['ssl']['ca_file'] }}
+ETCD_CERT_FILE={{ pillar['ssl']['crt_file'] }}
+ETCD_KEY_FILE={{ pillar['ssl']['key_file'] }}
+ETCD_TRUSTED_CA_FILE={{ pillar['ssl']['ca_file'] }}
 
-ETCD_PEER_CA_FILE={{ pillar['paths']['ca_dir'] }}/{{ pillar['paths']['ca_filename'] }}
-ETCD_PEER_CERT_FILE=/etc/pki/minion.crt
-ETCD_PEER_KEY_FILE=/etc/pki/minion.key
-ETCD_PEER_TRUSTED_CA_FILE={{ pillar['paths']['ca_dir'] }}/{{ pillar['paths']['ca_filename'] }}
+ETCD_PEER_CA_FILE={{ pillar['ssl']['ca_file'] }}
+ETCD_PEER_CERT_FILE={{ pillar['ssl']['crt_file'] }}
+ETCD_PEER_KEY_FILE={{ pillar['ssl']['key_file'] }}
+ETCD_PEER_TRUSTED_CA_FILE={{ pillar['ssl']['ca_file'] }}
 # ETCD_PEER_CLIENT_CERT_AUTH=on
```

Comment:
I disagree, bsc#1044590 is now effectively invalid (based on learning more about how etcd in proxy mode functions since the ticket was filed).

Fixing the authentication problem is just uncommenting those two variables and ensuring etcd's TLS has been correctly configured, as you have already added the client side parts of mutual TLS to K8S and flannel in this PR!